### PR TITLE
Set notification namespace for NotificationConfiguration

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Set notification namespace for NotificationConfigurtion

## Motivation and Context
Set notification namespace for NotificationConfigurtion

## Regression
No

## How Has This Been Tested?
Test using the reproducer https://github.com/minio/minio-java/issues/710 - using SetBucketNotifications.java example from minio-java

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :x: [more...](#afc40c5a-0cec-4fba-b8c3-5df46cdfe3e9) |
| mint-dist-xl.sh | :x: [more...](#724d9501-baf9-4ff9-8b89-4a92a8e765ec) |
<a name="afc40c5a-0cec-4fba-b8c3-5df46cdfe3e9"></a>
### 6789-5243351/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32325
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 570e52d21759:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 26 seconds
(2/13) Running aws-sdk-go tests ... done in 0 seconds
(3/13) Running aws-sdk-java tests ... done in 6 seconds
(4/13) Running aws-sdk-php tests ... done in 42 seconds
(5/13) Running aws-sdk-ruby tests ... done in 5 seconds
(6/13) Running mc tests ... done in 1 minutes and 4 seconds
(7/13) Running minio-dotnet tests ... FAILED in 3 minutes and 18 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Minio API responded with message=Please reduce your request)) ---> System.AggregateException: One or more errors occurred. (Minio API responded with message=Please reduce your request) ---> Minio.Exceptions.MinioException: Minio API responded with message=Please reduce your request
   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 363
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 505
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 351
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 483
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 229
   at Minio.Functional.Tests.FunctionalTest.PutObject_Task(MinioClient minio, String bucketName, String objectName, String fileName, String contentType, Int64 size, Dictionary`2 metaData, MemoryStream mstream) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 740
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.RemoveObjects_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 1738
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 218

Executed 6 out of 13 tests successfully.
```
<a name="724d9501-baf9-4ff9-8b89-4a92a8e765ec"></a>
### 6789-5243351/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32684
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0f6f58d0c80c:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 49 seconds
(2/13) Running aws-sdk-go tests ... done in 1 seconds
(3/13) Running aws-sdk-java tests ... done in 4 seconds
(4/13) Running aws-sdk-php tests ... done in 58 seconds
(5/13) Running aws-sdk-ruby tests ... done in 16 seconds
(6/13) Running mc tests ... done in 1 minutes and 16 seconds
(7/13) Running minio-dotnet tests ... FAILED in 5 minutes and 42 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.Fail failed. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.Fail failed. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.<>c.<ListIncompleteUpload_Test2>b__85_1(Exception ex) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2138
   at System.Reactive.AutoDetachObserver`1.OnErrorCore(Exception exception) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\AutoDetachObserver.cs:line 76
   at System.Reactive.AnonymousSafeObserver`1.OnError(Exception error) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\AnonymousSafeObserver.cs:line 62
   at System.Reactive.Linq.ObservableImpl.Throw`1._.Invoke() in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Linq\Observable\Throw.cs:line 42
   at System.Reactive.Concurrency.Scheduler.Invoke(IScheduler scheduler, Action action) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Concurrency\Scheduler.Simple.cs:line 83
   at System.Reactive.Producer`2.SubscribeRaw(IObserver`1 observer, Boolean enableSafeguard) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\Producer.cs:line 127
   at System.Reactive.Linq.QueryLanguage.<>c__DisplayClass330_0`1.<Create>b__0(IObserver`1 observer) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Linq\QueryLanguage.Creation.cs:line 46
   at System.Reactive.AnonymousObservable`1.SubscribeCore(IObserver`1 observer) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\AnonymousObservable.cs:line 37
   at System.Reactive.ObservableBase`1.ScheduledSubscribe(IScheduler _, AutoDetachObserver`1 autoDetachObserver) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\ObservableBase.cs:line 82
   at System.Reactive.Concurrency.CurrentThreadScheduler.Trampoline.Run(SchedulerQueue`1 queue) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Concurrency\CurrentThreadScheduler.cs:line 126
   at System.Reactive.Concurrency.CurrentThreadScheduler.Schedule[TState](TState state, TimeSpan dueTime, Func`3 action) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Concurrency\CurrentThreadScheduler.cs:line 94
   at System.Reactive.ObservableBase`1.Subscribe(IObserver`1 observer) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\ObservableBase.cs:line 51
   at Minio.Functional.Tests.FunctionalTest.ListIncompleteUpload_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2136
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 241

Executed 6 out of 13 tests successfully.
```

**Review by sinhaashish — DISMISSED** [review]
LGTM . 
1. Tested  by setting the notification for Elastic Search and then the `getBucketNotification` from minio java. 
2. Tested with `setBucketNotification` and `getBucketNotification` from minio java SDK.
Both worked fine.

File: cmd/bucket-notification-handlers.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ func (api objectAPIHandlers) GetBucketNotificationHandler(w http.ResponseWriter,
 		nConfig = &event.Config{}
```
Comment (kannappanr) [inline cmd/bucket-notification-handlers.go:83]: Can `nConfig = &event.Config{}` be changed to `nConfig = &event.Config{XMLNS: "http://s3.amazonaws.com/doc/2006-03-01/"}` ? and move the added code to `ParseConfig` function in `config.go` https://github.com/minio/minio/blob/master/pkg/event/config.go#L279. 

If the code is moved to `ParseConfig` function, the `notification.xml` stored in the backend will have the attribute saved. 
Also, with this change `readNotificationConfig` function will return a complete `config` structure and you do not have to set `XMLNS` field's value again.

Comment (kannappanr) [reply cmd/bucket-notification-handlers.go]: Move this added code to `ParseConfig` function in `config.go` https://github.com/minio/minio/blob/master/pkg/event/config.go#L279

Comment (kannappanr) [reply cmd/bucket-notification-handlers.go]: `SetBucketNofication.java` is failing. It looks like something that needs to be fixed on the `minio-java` side though

Comment (harshavardhana) [reply cmd/bucket-notification-handlers.go:83]: The explicit setting is on purpose @kannappanr - there is no reason to save them with a namespace, we save it with the way client sent it i.e without namespace. We only return it back with a namespace.

I don't see a good reason to do what you are suggesting.

**Comment by harshavardhana** [comment]
@sinhaashish @kannappanr rebased to master PTAL

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :x: [more...](#667afeba-8d15-48a1-83dd-2fc2b24cbefc) |
<a name="667afeba-8d15-48a1-83dd-2fc2b24cbefc"></a>
### 6789-256ff6b/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32564
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8a2f2f920f69:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 24 seconds
{
  "name": "awscli",
  "duration": 14944,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#9e56e749-5fa0-4ad0-a27a-0362ebd2ad39) |
<a name="9e56e749-5fa0-4ad0-a27a-0362ebd2ad39"></a>
### 6789-fa72615/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31502
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6b33636600ec:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 53 seconds
(2/13) Running aws-sdk-go tests ... done in 4 seconds
(3/13) Running aws-sdk-java tests ... done in 8 seconds
(4/13) Running aws-sdk-php tests ... done in 1 minutes and 15 seconds
(5/13) Running aws-sdk-ruby tests ... done in 41 seconds
(6/13) Running mc tests ... done in 2 minutes and 22 seconds
(7/13) Running minio-dotnet tests ... FAILED in 6 minutes and 45 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (One or more errors occurred. (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.)) ---> System.AggregateException: One or more errors occurred. (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) (Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.) ---> Minio.Exceptions.InternalClientException: Minio API responded with message=Unsuccessful response from server without XML error: The operation has timed out.
   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 450
   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 505
   at Minio.MinioClient.ExecuteTaskAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 351
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, String uploadId, Int32 partNumber, Byte[] data, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 483
   at Minio.MinioClient.PutObjectAsync(String bucketName, String objectName, Stream data, Int64 size, String contentType, Dictionary`2 metaData, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/ObjectOperations.cs:line 229
   at Minio.Functional.Tests.FunctionalTest.PutObject_Task(MinioClient minio, String bucketName, String objectName, String fileName, String contentType, Int64 size, Dictionary`2 metaData, MemoryStream mstream) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 740
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.RemoveObjects_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 1738
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 218

Executed 6 out of 13 tests successfully.
```

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by sinhaashish — APPROVED** [review]

**Pull request merged**